lib/rack/mock_request.rb

```diff
@@ -144,20 +143,25 @@ def self.env_for(uri = "", opts = {})
         end
       end
 
-      opts[:input] ||= String.new
+      unless opts.key?(:input)
```

Comment:
If we are switching from must to may, it seems a better default would be to not set an empty string if `:input` is not provided.

Reply:
I'm fine with that but it breaks compatibility since the default changes. Let me see how tricky it is.

Reply:
I've set the default to be nil and fixed the failing tests, so you can see the impact as a separate commit.